### PR TITLE
Avoid change always convert identifier_in to string in route53

### PR DESCRIPTION
**Pull request by cinek810**
rebase for #2477, ready_for_review

This is really important bugfix

### CODE REVIEW

**Comment by cinek810** [comment]
@Etherdaemon  maybe you can make community_review of this pull request? It solves #2501, is very simple and it's still not merged.

File: cloud/amazon/route53.py [hunk 1]
```diff
@@ -391,7 +391,10 @@ def main():
         #Need to save this changes in rset, because of comparing rset.to_xml() == wanted_rset.to_xml() in next block
 	rset.name = decoded_name
 
-        if rset.type == type_in and decoded_name.lower() == record_in.lower() and str(rset.identifier) == str(identifier_in):
+        if identifier_in is not None:
+            identifier_in = str(identifier_in)
+	
+        if rset.type == type_in and decoded_name.lower() == record_in.lower() and rset.identifier == identifier_in:
```
Comment (jjshoe) [inline cloud/amazon/route53.py:397]: Curious why str(rset.identifier) was dropped? I see what you did with identifier_in, and that makes sense.

Comment (cinek810) [inline cloud/amazon/route53.py:397]: rset.identifier is already a string or NoneType, so another str() is not needed

**Comment by gregdek** [comment]
Hmm, looks like this one got lost in the shuffle a bit -- kicking.

**Comment by gregdek** [comment]
Thanks @cinek810. To the current maintainers, @bpennypacker please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by bpennypacker** [comment]
shipit

**Comment by gregdek** [comment]
Thanks again to @cinek810 for this PR, and thanks @bpennypacker for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request merged**